SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/VisibilityTest/VisibilityTest.cpp

```diff
@@ -0,0 +1,336 @@
+			{
+					Counter++;
+			}
+			AssertEqual_Int(Counter, ExpectedReplicatedActors, TEXT("Number of TestHiddenActors in the server world"), NetTest);
+			FinishStep();
+		});
+	}
+
+	{	// Step 9 - Observe if the test actor has been replicated to the Clients.
+		AddStep(TEXT("ClientCheckReplicatedActorsAfterSetActorHidden"), FWorkerDefinition::AllClients, nullptr, [this](ASpatialFunctionalTest* NetTest)
+		{
+			int Counter = 0;
+			int ExpectedReplicatedActors = 0;
+			for (TActorIterator<AReplicatedVisibilityTestActor> Iter(GetWorld()); Iter; ++Iter)
+			{
+				Counter++;
+			}
+			AssertEqual_Int(Counter, ExpectedReplicatedActors, TEXT("Number of TestHiddenActors in the client world"), NetTest);
+			FinishStep();
+		});
+	}
+
+	{	// Step 10 - Server moves Client 1 close to the cube.
+		AddStep(TEXT("ServerMoveClient1CloseToCube"), FWorkerDefinition::Server(1), nullptr, [this](ASpatialFunctionalTest* NetTest)
+		{
+			ASpatialFunctionalTestFlowController* FlowController = GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
+			checkf(FlowController, TEXT("Can't be running test without valid FlowControl."));
+			APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+			ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+			if (PlayerCharacter->GetActorLocation().Equals(CharacterRemoteLocation, 50.0f))
+			{
+				if (PlayerCharacter->SetActorLocation(Character1StartingLocation))
+				{
+					if (PlayerCharacter->GetActorLocation().Equals(Character1StartingLocation, 50.0f))
+					{
+						FinishStep();
+					}
+				}
+			}
+		});
+	}
+
+	{   // Step 11 - Make sure that the Client moved to the correct location.
+		AddStep(TEXT("ClientCheckSecondMovement"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+		{
+			ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
+			checkf(FlowController, TEXT("Can't be running test without valid FlowControl."));
+
+			APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+			ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+			if (IsValid(PlayerCharacter))
+			{
+				if (PlayerCharacter->GetActorLocation().Equals(Character1StartingLocation, 50.0f))
+				{
+					FinishStep();
+				}
+			}
+		}, 5.0f);
+	}
+
+	{	// Step 12 - Observe if the test actor has been replicated to the clients.
+		AddStep(TEXT("ClientCheckFinalReplicatedActors"), FWorkerDefinition::AllClients, nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+		{
+			int Counter = 0;
+			int ExpectedReplicatedActors = 0;
+			for (TActorIterator<AReplicatedVisibilityTestActor> Iter(GetWorld()); Iter; ++Iter)
+			{
+				Counter++;
+				AssertEqual_Bool(Iter->IsHidden(), true, TEXT("TestHiddenActors Hidden in the client world"), NetTest);
```

Comment:
Since the number of expected actors here is 0 and this line is checking that any found actor has `IsHidden` doesn't that make this assert wrong? 
Also, do hidden actors actually get deleted from a clients view then?

Reply:
Do you think I should remove it then? there is no point checking that the actor `IsHidden` if there is not supposed to be there. I will remove that check